aodndata/moorings/products_handler.py

```diff
@@ -79,23 +84,21 @@ def preprocess(self):
                                                                     filter=ogc_filter,
                                                                     propertyname=['url', 'variables']
                                                                     )
```

Comment:
I would explicitly check for wfs_response['features'] being not empty instead of using hidden logic in a dict comprehension using a for loop. Anyway, you need to bail out eventually if no features are available - which the code is not doing.

```
url = f['properties']['url']
vars = f['properties']['variables'].split(', ')
feat = wfs_response['features']
if feat:
	self.input_file_variables = {url:vars};
else:
	raise...
```

Reply:
Done